### DISCUSSION TITLE
Question about the Sounds

### DISCUSSION THREAD

**Discussion by jorgegv** [jorgegv/rage1#42] [Q&A] [answered]
In issue #40, fraespe asked:

> Hi Jorge,
> 
> I have a simple question about your Sound system.
> Why haven't you used the beepfx library (there are available into the z88dk) ? Instead that, you have extrated the asm code and you have included into your game. What's the benefict ?
> 
> thanks in advance.

**Comment by jorgegv** [comment]
Hi,

At first I copied the asm version from the Churrera source, and all worked fine. Then I discovered that BeepFX was included in the standard Z88DK install and tried to use it but the sound was not good. This was because the library code is normally linked in lower addresses by Z88DK, and your program in higher addresses, so the sound code was being run from contended memory, and sound effects sounded bad.

So I rolled back and included again beepfx asm code as part of my engine, so that it is linked in upper memory for better sound. The beep_fx player code is just 148 bytes, so I decided it was a good tradeoff.

I will probably will come back to this when I start working on issue #4 , I'll have to deep dive in memory management for that one (it's related to 128K memory management).

J.

**Comment by fraespre** [comment, marked as answer]
ok, thanks Jorge for your early response.
I see that you have taken quite seriusly this development :-p

I take advance this communication to let you other related question. Would it have sense to use a unique engine for music and fx sound? it would have a big memory saving. Do you know about something similar to this idea ?

**Reply by jorgegv** [reply]
well I guess so... if your game has music. Given that the 48K spectrum is severely limited for playing music during the game, my first try would be to have some music only during the menu.

Music is secondary for me (for the moment). But yes, if I were to include music with my game, I would try tu use the same engine for both music and sound effects.

There are several 1-bit players out there which could be included... You can follow this link if interested: http://shiru.untergrund.net/software.shtml
